src/aztec3/circuits/kernel/private/c_bind.test.cpp

```diff
@@ -0,0 +1,111 @@
+
+using aztec3::circuits::apps::FunctionExecutionContext;
+using aztec3::circuits::apps::test_apps::escrow::deposit;
+
+} // namespace
+
+namespace aztec3::circuits::kernel::private_kernel {
+
+TEST(private_kernel, hash_tx_request)
+{
+    // Most of this is just setup (creating TxRequest to then be hashed)
+    const NT::address escrow_contract_address = 12345;
+
+    const NT::fr msg_sender_private_key = 123456789;
+    const NT::address msg_sender =
+        NT::fr(uint256_t(0x01071e9a23e0f7edULL, 0x5d77b35d1830fa3eULL, 0xc6ba3660bb1f0c0bULL, 0x2ef9f7f09867fd6eULL));
+    const NT::address tx_origin = msg_sender;
+
+    Composer deposit_composer = Composer("../barretenberg/cpp/srs_db/ignition");
+    DB db;
+
+    FunctionSignature<NT> function_signature{
+        .function_encoding = 1, // TODO: deduce this from the contract, somehow.
+        .is_private = true,
+        .is_constructor = false,
+    };
+
+    CallContext<NT> call_context{
+        .msg_sender = msg_sender,
+        .storage_contract_address = escrow_contract_address,
+        .tx_origin = msg_sender,
+        .is_delegate_call = false,
+        .is_static_call = false,
+        .is_contract_deployment = false,
+        .reference_block_num = 0,
+    };
+
+    NativeOracle deposit_oracle =
+        NativeOracle(db, escrow_contract_address, function_signature, call_context, msg_sender_private_key);
+    OracleWrapper deposit_oracle_wrapper = OracleWrapper(deposit_composer, deposit_oracle);
+
+    FunctionExecutionContext deposit_ctx(deposit_composer, deposit_oracle_wrapper);
+
+    auto amount = NT::fr(5);
+    auto asset_id = NT::fr(1);
+    auto memo = NT::fr(999);
+
+    OptionalPrivateCircuitPublicInputs<NT> opt_deposit_public_inputs = deposit(deposit_ctx, amount, asset_id, memo);
+    PrivateCircuitPublicInputs<NT> deposit_public_inputs = opt_deposit_public_inputs.remove_optionality();
+
+    TxRequest<NT> tx_request = TxRequest<NT>{
+        .from = tx_origin,
+        .to = escrow_contract_address,
+        .function_signature = function_signature,
+        .args = deposit_public_inputs.args,
+        .nonce = 0,
+        .tx_context =
+            TxContext<NT>{
+                .is_fee_payment_tx = false,
+                .is_rebate_payment_tx = false,
+                .is_contract_deployment_tx = false,
+                .contract_deployment_data = ContractDeploymentData<NT>(),
+                .reference_block_num = 0,
+            },
+        .chain_id = 1,
+    };
+
+    // Perform c_bind hash and check result
+    std::vector<uint8_t> buf;
+    write(buf, tx_request);
+    uint8_t* output = (uint8_t*)malloc(sizeof(uint8_t) * 32);
+    // Make the c_bind call to the hash tx request
+    private_kernel__hash_tx_request(buf.data(), output);
+
+    NT::fr hash = NT::fr::serialize_from_buffer(output);
+
+    std::string got_hash = format(hash);
+    std::string expected_hash = "0x0fb3576464a061dd1322b9d47a18f470d1a4405e076a20f86c87bfdcd59c789f";
```

Comment:
Instead of hard-coding this, can we do: `tx_request.hash()->to_buffer()` and compare that with `c_bind`'s result? We would want to make the tests agnostic to changes to the hash function algorithm.

Reply:
Does this suffice? https://github.com/AztecProtocol/aztec3-circuits/pull/26/files#diff-14f002b1c41e9ac3c4fbb7a0991d2325b91f189442919413303740f6384e3e5fR108

I guess I could do `tx_request.hash()->to_buffer()` instead of doing `serialize_from_buffer` on the cbind-generated output. Which do you think is the better way?